auth/method.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import "fmt"
+
+// Method is the enumeration of authentication methods supported by Veraison
+// service. It implements the pflag.Value interface.
+type Method string
+
+const (
+	MethodPassthrough Method = "passthrough"
+	MethodBasic       Method = "basic"
+	MethodOauth2      Method = "oauth2"
+)
+
+// String representation of the Method
+func (o *Method) String() string {
+	return string(*o)
+}
+
+// Set the value of the Method
+func (o *Method) Set(v string) error {
+	switch v {
+	case "none", "passthrough":
+		*o = MethodPassthrough
+	case "basic":
+		*o = MethodBasic
+	case "oauth2":
+		*o = MethodOauth2
+	default:
+		return fmt.Errorf("unexpected Method %q", v)
+	}
+
+	return nil
+}
+
+// Type returns the string representing the type name (used by pflag).
+func (o *Method) Type() string {
```

Comment:
This is not clear to me the purpose and scope of this method

Reply:
it's a method required by the [`pflag.Value`](https://pkg.go.dev/github.com/spf13/pflag#Value) interface, which the `auth.Method` type implements

Reply:
We should clarify a bit more in the comment to make it better!

Reply:
The comment already states that it's used by `pflag`